plugins/scaffolder-backend/src/scaffolder/stages/publish/types.ts

```diff
@@ -0,0 +1,26 @@
+/*
+ * Copyright 2020 Spotify AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { TemplateEntityV1alpha1 } from '@backstage/catalog-model';
+import { RequiredTemplateValues } from '../templater';
+import { JsonValue } from '@backstage/config';
+
+export type Publisher = {
+  publish(opts: {
+    entity: TemplateEntityV1alpha1;
+    values: RequiredTemplateValues & Record<string, JsonValue>;
```

Comment:
Think it would make sense to narrow this down to only the storePath?

Reply:
Yes and no. Like the fact that all the stages get all of the data. Dont wanna make this a special snowflake for only our implementation.